packages/context/src/notification/NotificationProvider.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, createContext, useEffect, useContext } from 'react';
+import { HttpTransportType, HubConnection, HubConnectionBuilder, LogLevel } from '@microsoft/signalr/dist/esm/index.js';
+import { useApiServices } from '../api/index.js';
+
+export type NotificationConnectionInfo = {
+    url: string;
+    accessToken: string;
+};
+
+export type Subscription<T> = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subscribe: (topic: string, callback: (...args: T[]) => any) => void;
+    unsubscribe: (topic: string, callback?: (...args: T[]) => void) => void;
+};
+
+export type DocumentChange = {
+    objectId: string;
+    instanceId: string;
+    documentId: string;
+    type: string;
+};
+
+export type InstanceChange = string;
+
+export type NotificationContextType = {
+    documentChanges?: Subscription<DocumentChange>;
+    instanceChanges?: Subscription<InstanceChange>;
+};
+
+export const NotificationContext = createContext<NotificationContextType>({});
+
+NotificationContext.displayName = 'NotificationContext';
+
+function NotificationProvider({ children }: { children: React.ReactNode }) {
+    const [instancesNotification, setInstancesNotification] = useState<HubConnection>();
+    const [documentsNotification, setDocumentsNotification] = useState<HubConnection>();
+
+    const api = useApiServices();
+
+    useEffect(() => {
+        const getConnectionInfo = (hubName: string) => {
+            return api.post<NotificationConnectionInfo>(`/notification/hubs/${hubName}/negotiate`);
+        };
+
+        const getConnection = async () => {
+            try {
+                const instancesConnectionInfo = await getConnectionInfo('instanceChanges');
+                const documentsConnectionInfo = await getConnectionInfo('documentChanges');
+
+                if (instancesConnectionInfo) {
+                    const options = {
+                        accessTokenFactory: async () => {
+                            if (instancesConnectionInfo.accessToken) {
+                                return instancesConnectionInfo.accessToken;
+                            } else {
+                                return getConnection();
```

Comment:
I assume `accessTokenFactory` is supposed to return a token, but `getConnection` doesn't return anything.